cg/services/orders/store_order_services/store_fastq_order_service.py

```diff
@@ -81,56 +55,49 @@ def create_maf_case(self, sample_obj: Sample, order: Order) -> None:
-            customer_internal_id=customer_id
+            customer_internal_id=order.customer
         )
-        if not customer:
-            raise OrderError(f"Unknown customer: {customer_id}")
         new_samples = []
         case: Case = self.status_db.get_case_by_name_and_customer(
             customer=customer, case_name=ticket_id
         )
-        submitted_case: dict = items[0]
         status_db_order = Order(
             customer=customer,
             order_date=datetime.now(),
             ticket_id=int(ticket_id),
         )
         with self.status_db.session.no_autoflush:
-            for sample in items:
+            for sample in order.samples:
                 new_sample = self.status_db.add_sample(
-                    name=sample["name"],
-                    sex=sample["sex"] or "unknown",
-                    comment=sample["comment"],
-                    internal_id=sample.get("internal_id"),
-                    order=order,
-                    ordered=ordered,
+                    name=sample.name,
+                    sex=sample.sex or "unknown",
+                    comment=sample.comment,
+                    internal_id=sample._generated_lims_id,
+                    order=order.name,
+                    ordered=datetime.now(),
                     original_ticket=ticket_id,
-                    priority=sample["priority"],
-                    tumour=sample["tumour"],
-                    capture_kit=sample["capture_kit"],
-                    subject_id=sample["subject_id"],
+                    priority=sample.priority,
+                    tumour=sample.tumour,
+                    capture_kit=sample.capture_kit,
+                    subject_id=sample.subject_id,
                 )
-                new_sample.customer: Customer = customer
-                application_tag: str = sample["application"]
+                new_sample.customer = customer
+                application_tag: str = sample.application
                 application_version: ApplicationVersion = (
-                    self.status_db.get_current_application_version_by_tag(tag=application_tag)
+                    self.status_db.get_current_application_version_by_tag(application_tag)
                 )
-                if application_version is None:
-                    raise OrderError(f"Invalid application: {sample['application']}")
-                new_sample.application_version: ApplicationVersion = application_version
+                new_sample.application_version = application_version
                 new_samples.append(new_sample)
                 if not case:
                     case = self.status_db.add_case(
-                        data_analysis=Workflow(submitted_case["data_analysis"]),
-                        data_delivery=DataDelivery(submitted_case["data_delivery"]),
+                        data_analysis=Workflow.RAW_DATA,
+                        data_delivery=DataDelivery.FASTQ,
```

Comment:
I remember making this change. To me using Workflow and DataDelivery makes sense here. Do you agree @islean?

Reply:
I think so too! Although we need to make sure that the storing service is not used for other workflows I guess. Potentially we should use the one sent on the order object

Reply:
I think this is fine for now though